### ISSUE TITLE
disambiguate functional stuff from modules that have the same name

### ISSUE THREAD

**Issue by wjessup** [ml-explore/mlx#178]
same documentation, but one is used as a module and one is functional. Caused issues in my code because capitalizations are different etc. 

https://ml-explore.github.io/mlx/build/html/python/_autosummary_functions/mlx.nn.relu.html
https://ml-explore.github.io/mlx/build/html/python/_autosummary/mlx.nn.ReLU.html

**Comment by wjessup** [comment]
would recommend adding something like 

'print(nn.relu(mx.array([-1,2])))' as an example for the functional relu documentation.

**Comment by awni** [comment]
I'm not following your comment. Can you be more specific about what the problem is here?

**Comment by wjessup** [comment]
I wrote some code like:

```
class DQN(nn.Module):
    def __init__(self, n_observations, n_actions):
        super().__init__()
        self.layer1 = nn.Linear(n_observations, 64)
        self.layer2 = nn.Linear(64, 64)
        self.layer3 = nn.Linear(64, n_actions)
        

    def __call__(self, x):
        x = nn.ReLU(self.layer1(x))
        x = nn.ReLU(self.layer2(x))
        return self.layer3(x)

```

But this will throw an error because nn.ReLU is different than nn.relu. The documentation doesn't clarify if/how they are different. 

The capitalized ReLU is used something like:

```
class DQN(nn.Module):
    def __init__(self, n_observations, n_actions):
        super().__init__()
        #why are the layers ordered like this?
        self.layer1 = nn.Linear(n_observations, 64)
        self.layer2 = nn.Linear(64, 64)
        self.layer3 = nn.Linear(64, n_actions)
        self.my_relu_layer = nn.ReLU()
        

    def __call__(self, x):
        x = self.layer1(x)
        x = self.my_relu_layer(x)
        x = self.layer2(x)
        x = self.my_relu_layer(x)
        return self.layer3(x)
```

Not sure why you'd want a stateless / functional bit of code structured like this.

What would the impact be of removing these from: https://github.com/ml-explore/mlx/blob/e28b57e37194556b681c6bf044201f8a2b3e1cb5/python/mlx/nn/layers/activations.py#L210C1-L212C9

**Comment by awni** [comment]
Some people find the functional activations useful some prefer the modules. In general I agree it's a bit messy to have both, I would just have the functional since those are the most flexible. Will update the docs for now.

**Comment by wjessup** [comment]
If it's simply preference or habit, there is a benefit to simplifying to the functional approach only imo. Would reduce the code complexity, maintenance footprint, documentation while improving end user clarity / mental model. 

I'm currently porting a torch project to MLX and it's not exactly like changing np to jax.numpy. A lot of work is required and more clarity in 'the one way to do it' would actually help.

**Comment by awni** [comment]
Addressed in #198
